xmtp_mls/src/groups/mls_sync.rs

```diff
@@ -934,11 +951,19 @@ where
                     let maybe_validated_commit = self.stage_and_validate_intent(&mls_group, &intent, provider, &message, envelope).await;
 
                     provider.transaction(|provider| {
-                        let is_updated =
-                            provider
+                        let requires_processing = if allow_cursor_increment {
+                            provider.conn_ref().update_cursor(
+                                &envelope.group_id,
+                                EntityKind::Group,
+                                cursor as i64,
+                            )?
+                        } else {
+                            let current_cursor = provider
                                 .conn_ref()
-                                .update_cursor(&envelope.group_id, EntityKind::Group, cursor as i64)?;
```

Comment:
I believe the same concerns around `update_cursor` from stream for external message processing also apply when processing own messages, so I added the check here as well in the `match intent`/ "_own message_" side of this function.